lib/spack/spack/cmd/make_installer.py

```diff
@@ -0,0 +1,137 @@
+            contents += line_to_rtf(line)
+    return rtf_header.format(contents)
+
+
+def setup_parser(subparser):
+    spack_source_group = subparser.add_mutually_exclusive_group(required=True)
+    spack_source_group.add_argument(
+        '-v', '--spack_version', default="",
+        help='download given spack version e.g. 0.16.0')
+    spack_source_group.add_argument(
+        '-s', '--spack_source', default="",
+        help='full path to spack source')
+
+    subparser.add_argument(
+        '-g', '--git-installer-verbosity', default="",
+        choices=set(['SILENT', 'VERYSILENT']),
+        help="Level of verbosity provided by bundled Git Installer.\
+             Default is fully verbose",
+        required=False, action='store', dest="git_verbosity"
+    )
+
+    subparser.add_argument(
+        'output_dir', help="output directory")
+
+
+def make_installer(parser, args):
+    """
+       Use CMake to generate WIX installer in newly created build directory
+    """
+    if sys.platform == 'win32':
+        output_dir = args.output_dir
+        cmake_spec = Spec('cmake')
+        cmake_spec.concretize()
+        cmake_path = os.path.join(cmake_spec.prefix, "bin", "cmake.exe")
+        cpack_path = os.path.join(cmake_spec.prefix, "bin", "cpack.exe")
+        spack_source = args.spack_source
+        git_verbosity = ""
+        if args.git_verbosity:
+            git_verbosity = "/" + args.git_verbosity
+
+        if spack_source:
+            if not os.path.exists(spack_source):
+                print("%s does not exist" % spack_source)
+                return
+            else:
+                if not os.path.isabs(spack_source):
+                    spack_source = posixpath.abspath(spack_source)
+                spack_source = spack_source.replace('\\', '/')
+
+        spack_version = args.spack_version
+
+        here = os.path.dirname(os.path.abspath(__file__))
+        source_dir = os.path.join(here, "installer")
+        posix_root = spack.paths.spack_root.replace('\\', '/')
+        spack_license = posixpath.join(posix_root, "LICENSE-APACHE")
+        rtf_spack_license = txt_to_rtf(spack_license)
+        spack_license = posixpath.join(source_dir, "LICENSE.rtf")
+
+        with open(spack_license, 'w') as rtf_license:
+            written = rtf_license.write(rtf_spack_license)
+            if written == 0:
+                raise RuntimeError("Failed to generate properly formatted license file")
+        spack_logo = posixpath.join(posix_root,
+                                    "share/spack/logo/favicon.ico")
+
+        try:
+            subprocess.check_call(
+                ('"%s" -S "%s" -B "%s" -DSPACK_VERSION=%s '
+                 '-DSPACK_SOURCE="%s" -DSPACK_LICENSE="%s" '
+                 '-DSPACK_LOGO="%s" -DSPACK_GIT_VERBOSITY="%s"')
+                % (cmake_path, source_dir, output_dir, spack_version, spack_source,
+                   spack_license, spack_logo, git_verbosity),
+                shell=True)
```

Comment:
Can you use the `Executable` object provided by Spack (which should be able to `shlex.split` and avoid use of `shell=true`).

Reply:
And I should clarify to say it would be good to replace any call to `subprocess.check_call` and to avoid `shell=True` universally).